test/Mono.Linker.Tests.Cases/Reflection/ObjectGetType.cs

```diff
@@ -1458,5 +1461,44 @@ public static void Test ()
 				}
 			}
 		}
+
+		[Kept]
+		class NullValue
+		{
+			[Kept]
+			class TestType
+			{
+			}
+
+			[Kept]
+			[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresAll) + "(Type)", nameof (Object.GetType) + "()")]
+			public static void Test ()
+			{
+				TestType nullInstance = null;
+				// Even though this throws at runtime, we warn about the return value of GetType
+				nullInstance.GetType ().RequiresAll ();
+			}
+		}
+
+		[Kept]
+		class NoValue
+		{
+			[Kept]
+			class TestType
+			{
+			}
+
+			[Kept]
+			static TestType GetInstance () => null;
+
+			[Kept]
+			[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresAll) + "(Type)", nameof (Object.GetType) + "()")]
+			public static void Test ()
+			{
+				TestType noValue = GetInstance ();
+				// Even though this throws at runtime, we warn about the return value of GetType
+				noValue.GetType ().RequiresAll ();
```

Comment:
This one I think is correct - we don't really track values across method calls (I mean in theory this could be handled through const propagation, but that's a different problem). So the fact that we warn here is a good thing I think.

Reply:
Thanks for catching this, the test wasn't exactly what I wanted. I will change it to use the same pattern for getting an empty value (of type Type) as in the other tests.